# Import sth

In [4]:
import math
import matplotlib.pyplot as plt

# Boundary condition

## Air inlet velocity for 

In [2]:
# to compute the inlet velocity of air
diammeter_fuel_inner = 4*1e-3
diammeter_fuel_outer = 24*1e-3
diammeter_air_inner = 70*1e-3
airflow_section_area = 3.14*(diammeter_air_inner**2 - diammeter_fuel_outer**2)/4

airflow_volume_speed = 0.000747329
 #0.00597863

airflow_velocity_speed = airflow_volume_speed/airflow_section_area
print(f'the inlet velocity of air is :{airflow_velocity_speed} m/s')

the inlet velocity of air is :0.2201691639611824 m/s


## Fuel inlet velocity for fluent input

In [136]:
# to compute the fuel inlet velocity of air

fuelflow_section_area = 3.14*diammeter_fuel_inner**2 /4

fuelflow_volume_speed = 0.0002512

fuelflow_velocity_speed = fuelflow_volume_speed/fuelflow_section_area
print(f'the velocity of fuel is :{fuelflow_velocity_speed} m/s')

the velocity of fuel is :20.0 m/s


# Swril number (Sn)

## Paper validation of swirl number

|Swirl type| Diameter (mm) | Vane angel(°) |Vane number|Swirl number|
|---|-----|-------|-----------|-----------|
| Type 1 | Inner: 13, outer 34.4|45 | 12 | 0.736 |
| Type 2 | -|60 | 12 | 1.27 |

In [57]:
# refer paper: {Hayakawa, 2017 #1003} Formula (1), Table 1 - Swirler specifications

# input the diameter and angle of the swirl vane
vane_angle = 60 #in angle
#angle_in_radian = vane_angle/360*(2*math.pi) # angle in radian
vane_angle_in_radian = math.radians(vane_angle) # angle in radian
number_of_vane = 6 # 叶片数（并不参与计算）
diameter_of_inner = 13 # d_h, the hub diameter in mm
diameter_of_outer = 34.4 # d_t, the swirl tip diameter in mm
d_ratio = diameter_of_inner/diameter_of_outer

# formula
swirl_number = (2/3)*((1-d_ratio**3)/(1-d_ratio**2))*math.tan(vane_angle_in_radian)
swirl_number_round = round(swirl_number,3) # Round a number to only three decimals

# output
print(f"彼时\n旋流器内、外直径大小为：{diameter_of_inner}、{diameter_of_outer} （mm）；\n角度及弧度大小为：{vane_angle} (°)、{vane_angle_in_radian} (rad)\n旋流数为(Sn): {swirl_number_round}")

彼时
旋流器内、外直径大小为：13、34.4 （mm）；
角度及弧度大小为：60 (°)、1.0471975511965976 (rad)
旋流数为(Sn): 1.274


In [46]:
# 角度转换弧度
a = 45 # 角度
in_radians = 45/360*2*3.14
print(in_radians)
print(math.radians(45))

0.785
0.7853981633974483


甲烷掺氢旋流钝体之——旋流结构参数
|Swirl type| Diameter (mm) | Vane angel(°) |Vane number|Swirl number|
|---|-----|-------|-----------|-----------|
| Swirl 1 | inner: 25, outer: 70 |30 | 6 | xxx |
| Swirl 2 | - | 45 | 6 | xxx |
| Swirl 3 | - | 60 | 6 | xxx |

In [58]:
# 暂定: 旋流器结构高度为20mm

# input the diameter and angle of the swirl vane
vane_angle = 75 #in angle
#angle_in_radian = vane_angle/360*(2*math.pi) # angle in radian
vane_angle_in_radian = math.radians(vane_angle) # angle in radian
number_of_vane = 6 # 叶片数（并不参与计算）
diameter_of_inner = 25 # d_h, the hub diameter in mm
diameter_of_outer = 70 # d_t, the swirl tip diameter in mm
d_ratio = diameter_of_inner/diameter_of_outer

# formula
swirl_number = (2/3)*((1-d_ratio**3)/(1-d_ratio**2))*math.tan(vane_angle_in_radian)
swirl_number_round = round(swirl_number,3) # Round a number to only three decimals

# output
print(f"彼时\n旋流器内、外直径大小为：{diameter_of_inner}、{diameter_of_outer} （mm）；\n角度及弧度大小为：{vane_angle} (°)、{vane_angle_in_radian} (rad)\n计算旋流数大小为(Sn): {swirl_number_round}")

彼时
旋流器内、外直径大小为：25、70 （mm）；
角度及弧度大小为：60 (°)、1.0471975511965976 (rad)
计算旋流数大小为(Sn): 1.263


|Swirl type| Diameter (mm) | Vane angel(°) |Vane number|Swirl number|
|---|-----|-------|-----------|-----------|
| Swirl 1 | inner: 25, outer: 70 |30 | 6 | 0.421 |
| Swirl 2 | - | 45 | 6 | 0.729 |
| Swirl 3 | - | 60 | 6 | 1.263 |
| Swirl 4 | - | 75 | 6 | 2.722 |

甲烷掺氢旋流钝体之——钝体结构参数（待完善-2023.04.19）
| Bluff body(BB) type | Height (mm) | BB angel(°) | Vane number | Swirl number |
|---|-----|-------|-----------|-----------|
| BB 1 | inner: 25, outer: 70 |30 | 6 | xxx |
| BB 2 | - | 45 | 6 | xxx |
| BB 3 | - | 60 | 6 | xxx |

## 求旋流叶片角度theta

### 计算旋流数Sn

确定swirl vane结构参数

- 变量
    - 旋流叶片数量（并不影响旋流数Sn）
    - 旋流角度
    - 旋流数Sn

$$
Sn=\frac{2}{3}\left[ \frac{1-\left( D_i/D_o \right) ^3}{1-\left( D_i/D_o \right) ^2} \right] \tan \theta
$$



建模思路：
-  为计算旋流数Sn，需根据底部的直角边d_mid以及二分之一的螺距pitch，近似形成一个直角三角形，通过反正切三角函数，可以计算$\theta$角度；
$$
tan(\theta)=\frac{0.5Pitch}{D_{mid}}
$$
-  在底部的直角边$D_{mid}$确定的情况下，$\theta$角度及对应的$tan(\theta)$的值由螺距pitch所决定；
-  已知$tan(\theta)$的值，可以求得对应$\theta$下的pitch值（建模依据）
-  通过$\theta$角度代入旋流数Sn计算公式，计算得到所建3D模型的旋流数Sn；
    - A: 给定某一$\theta$角度（用于旋流叶片建模），求得对应的Sn
        -  建模：通过$\theta$以及正切函数$tan(\theta)$，计算得到对应的pitch值，用于旋流叶片建模；
        -  计算Sn：将$\theta$直接代入旋流数Sn计算公式
    - B: 给定对应的Sn，求对应的$\theta$角度（用于旋流叶片建模）
        -  建模：通过对应的Sn，求得$tan(\theta)$中的$\theta$值，

In [87]:
# A: 给定某一theta角度（用于旋流叶片建模），求得对应的Sn
# 情景：计算theta角度分别为60°，70°，80°条件下的Sn，此时：
# 1.根据tan(theta)公式可求出pitch的大小用于建模
# 2.代入Sn公式求得Sn的值

# ----
# 空气流道内直径(d_h, the hub diameter)(uinit: mm)
diameter_of_inner = 15
# 空气流道外直径(d_t, the swirl tip diameter)(uinit: mm)
diameter_of_outer = 30
d_ratio = diameter_of_inner/diameter_of_outer
# 计算已知的diameter_mid（底部的直角边）
diameter_mid = 0.5*(diameter_of_inner + diameter_of_outer)
# ----
# 需要确定的pitch（螺距高度：形成一圈螺纹需要多长的距离）(uinit: mm)

# 输入变量
# 🟢已知角度theta（旋流叶片的角度为vane angle；theta是为直角三角形中的标记）

theta_in_degree = 65  # 角度（单位为度）
# 将角度转换为弧度
theta_in_radian = math.radians(theta_in_degree)

# 计算对应角的正切值
tan_value = math.tan(theta_in_radian)
# 确定的pitch值，作为inventor旋流叶片coil命令下的键入值
pitch =  2 * diameter_mid / tan_value

# 计算旋流数Sn
swirl_number = (2/3)*((1-d_ratio**3)/(1-d_ratio**2))*tan_value

# 输出结果，Pitch值和旋流数Sn
print("旋流叶片角度为{}°，对应角的正切值为{:.2f}，对应的Pitch值为{:.2f}，对应的Sn值为{:.2f}".format(theta_in_degree, tan_value, pitch, swirl_number))

旋流叶片角度为65°，对应角的正切值为2.14，对应的Pitch值为20.98，对应的Sn值为1.67


  ## A的计算结果汇总
  
  | 结构1   | 叶片数n | 旋转叶片的角度$\theta$ | 旋流数Sn | Pitch (unit:mm)|
  | ------- | ------- | --------------- | -------- |-------- |
  | Swirl theta 1 | 5       | 30°     |   0.45   | 77.94 |
  | Swirl theta 2 | 5       | 35°     |   0.54   | 64.27 |
  | Swirl theta 3 | 5       | 40°     |   0.65   | 53.63 |
  | ✓Swirl theta 4 | 5       | 45°     |   0.78   | 45   |
  | Swirl theta 5 | 5       | 50°     |   0.93   | 37.76 |
  | ✓ Swirl theta 6 | 5       | 55°     |   1.11   | 31.51 |
  | Swirl theta 7 | 5       | 60°     |   1.35   | 25.98 |
  | ✓ Swirl theta 8 | 5       | 65°     |   1.67   | 20.98 |
  | Swirl theta 9 | 5       | 75°     |   2.9    | 12.06 |
  | Swirl Sn 1   | -       |    -        | 0.6    | - |
  | Swirl Sn 1   | -       |    -        | 0.7    | - |
  | Swirl Sn 1   | -       |    -        | 0.9    | - |

In [88]:
# B: 给定对应的Sn，求对应的$\theta$角度（用于旋流叶片建模）
# 情景：建模Sn分别为0.4，0.6，0.8条件下的3D旋流叶片，此时：
# 1.根据tan(theta)公式可求出pitch的大小用于建模
# 2.已知Sn的值

## Re计算


In [1]:
print("Hello world!")

Hello world!
